### ISSUE TITLE
Add length related assertions to StringAssert

### ISSUE THREAD

**Issue by georgberky** [joel-costigliola/assertj-core#1319]
#### Summary
String assertions often concern the string's size. Currently we would have to use something like

```java
assertThat("some string".length()).isGreaterThan(10);
```
this is not really fluent and allows for semantically incorrect checks like negative String lengths. Instead we would like to use something like this:

#### Example

```java
assertThat("some string").hasSizeLessThan(10); //< 10
assertThat("some string").hasSizeLessOrEqualTo(10); //<= 10
assertThat("some string").hasMaxSize(10); //<= 10

//analog methods for greater and greater or equals
```

**Comment by georgberky** [comment]
We are working on this issue at [Softwerkskammer Ruhrgebiet Hackergarten](https://www.meetup.com/Softwerkskammer-Ruhrgebiet/events/248952817/)

**Comment by martin-tarjanyi** [comment]
Hi @georgberky 
I guess, I'm a bit late with this but you might want to take a look at this PR:
https://github.com/joel-costigliola/assertj-core/pull/1285

This implements most of these methods and not just for String but for all iterables.

@joel-costigliola: any chance that you can review my PR. I know it's a huge one...

**Comment by joel-costigliola** [comment]
@martin-tarjanyi I will to but it's huge.

**Comment by joel-costigliola** [comment]
@martin-tarjanyi you're right, this slipped out of my mind, I will review it to but it's huge so I haven't got the mental energy to do it yet (I'll take care of merging/rebasing on master after this PR is merged).

**Comment by sparsick** [comment]
@martin-tarjanyi Strings are not iterable in Java, so that I think it will not have conflict with your pull request.

**Comment by martin-tarjanyi** [comment]
@sparsick 
I'm aware of that in core Java there is no relation between String and Iterables. But in AssertJ there is an [EnumerableAssert](https://github.com/joel-costigliola/assertj-core/blob/master/src/main/java/org/assertj/core/api/EnumerableAssert.java) class which is a common interface for the string, array, iterables and some other assertion classes.

In my PR I added the new methods to EnumerableAssert and I implemented the methods for all child classes.
